### DISCUSSION TITLE
How do I add a model ?

### DISCUSSION THREAD

**Discussion by anjanesh** [open-webui/open-webui#6835] [Q&A]
![CleanShot 2024-11-10 at 13 51 12](https://github.com/user-attachments/assets/9ab011a8-44fb-4371-b377-75ff0d88e7f7)

**Comment by parniantaghipour** [comment]
If you are already have connection to ollama you can type the name of the model and it suggest downloading the model.

You can check your connection in the admin panel.

**Comment by anjanesh** [comment]
When I enter llama3.2:3b my console says :

```
INFO  [open_webui.apps.openai.main] get_all_models()
ERROR [open_webui.apps.openai.main] Connection error: Cannot connect to host api.openai.com:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)')]
INFO  [open_webui.apps.ollama.main] get_all_models()
ERROR [open_webui.apps.ollama.main] Connection error: Cannot connect to host localhost:11434 ssl:default [Multiple exceptions: [Errno 61] Connect call failed ('::1', 11434, 0, 0), [Errno 61] Connect call failed ('127.0.0.1', 11434)]
INFO  [open_webui.apps.ollama.main] get_all_models()
ERROR [open_webui.apps.ollama.main] Connection error: Cannot connect to host localhost:11434 ssl:default [Multiple exceptions: [Errno 61] Connect call failed ('::1', 11434, 0, 0), [Errno 61] Connect call failed ('127.0.0.1', 11434)]
INFO  [open_webui.apps.ollama.main] url: http://localhost:11434
ERROR [asyncio] Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x16aadd190>
INFO:     127.0.0.1:62851 - "POST /ollama/api/pull/0 HTTP/1.1" 500 Internal Server Error
```

**Comment by flefevre** [comment]
At first you have to go to the admin panel, then 'connection' rab, to fill the form for the URL of your ollama or litellm inference model servers 
Here you can also ask to download a model.
Hope it helps.

**Reply by anjanesh** [reply]
So I need to [download](https://ollama.com/download) and install and run ollama before I can run open-webui ?

**Comment by flefevre** [comment]
Personaly I have followed the tips here

https://docs.openwebui.com/getting-started/quick-start/#advanced-configuration-connecting-to-ollama-on-a-different-server

https://openwebui.com/getting-started/using-openwebui/troubleshooting-ollama

**Comment by anjanesh** [comment]
I would like to avoid using Docker for now - is it not possible to connect to a ollama server on my laptop that's run not using docker ?

**Reply by flefevre** [reply]
Yes, you can add in the admin tab in the section 'connexion' an url on an ollama server setup with whatever you want (docker postman , cloud)

**Comment by TaggM** [comment]
The answer seems to be:
The Docker image for Open WebUI is bundled with an Ollama server URL of: localhost:11434 but, that is not visible to Dockerized Open WebUI. Instead, it needs an Ollama server URL of: host.docker.internal:11434
